app/models/concerns/account_interactions.rb

```diff
@@ -117,9 +117,16 @@ def block!(other_account, uri: nil)
                        .find_or_create_by!(target_account: other_account)
   end
 
-  def mute!(other_account, notifications: nil)
+  def mute!(other_account, notifications: nil, duration: 0)
     notifications = true if notifications.nil?
-    mute = mute_relationships.create_with(hide_notifications: notifications).find_or_create_by!(target_account: other_account)
+    mute = mute_relationships.find_by(target_account: other_account)
+    expires_at = (duration.zero? ? nil : (Time.current + duration.seconds))
+    if mute
+      mute.update!(expires_at: expires_at)
+    else
+      mute = mute_relationships.create_with(hide_notifications: notifications, expires_at: expires_at).find_or_create_by!(target_account: other_account)
+    end
```

Comment:
Thinking about it, you could use the `Expirable` concern in the `Mute` model and use `mute.expires_in = duration`, with `duration` maybe defaulting to `nil` rather than `0`.

Reply:
I changed to use ```Expireable``` concern in the ```Mute``` model.